### PR TITLE
[BACKLOG-37434] Move metastore locator plugin access out of the

### PR DESCRIPTION
**Pull request by peterrinehart**
constructor to ensure it isn't used before the plugin is loaded.

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 17m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-services-api
```
---
:ok_hand: All tests passed!
 
Tests run: **796**, Failures: **0**, Skipped: **2** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-hadoop-shims/job/PR-1359/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 15m 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-services-api
```
---
:ok_hand: All tests passed!
 
Tests run: **796**, Failures: **0**, Skipped: **2** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-hadoop-shims/job/PR-1359/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: common-services-api/src/main/java/org/pentaho/big/data/api/cluster/service/locator/impl/NamedClusterServiceLocatorImpl.java [hunk 1]
```diff
@@ -54,28 +54,34 @@ public class NamedClusterServiceLocatorImpl implements NamedClusterServiceLocato
...
-    metastoreLocator = metastoreLocator1;
     this.namedClusterManager = namedClusterManager;
     readWriteLock = new ReentrantReadWriteLock();
     serviceVendorTypeMapping = new HashMap<>();
   }
 
+  protected MetastoreLocator getMetastoreLocator() {
+    if ( this.metastoreLocator == null ) {
+      MetastoreLocator metastoreLocator1;
+      try {
```
Comment (abryant-hv) [inline common-services-api/src/main/java/org/pentaho/big/data/api/cluster/service/locator/impl/NamedClusterServiceLocatorImpl.java:73]: Same question as the other diff... this isn't thread safe, does it need to be?

**Review by abryant-hv — COMMENTED** [review]
See question in PR

File: common-services-api/src/main/java/org/pentaho/big/data/api/cluster/service/locator/impl/NamedClusterServiceLocatorImpl.java (hunk 1 above)
Comment (peterrinehart) [reply common-services-api/src/main/java/org/pentaho/big/data/api/cluster/service/locator/impl/NamedClusterServiceLocatorImpl.java:73]: No, this shouldn't need to be thread safe.

**Pull request closed without merging**